duckduckgo/duckduckgo-privacy-extension
# Move surrogates to web accessible resources

**Reviewer:** @dharb 



## Description:

1. Splits our single surrogate file into individual files in web_accessible_resources
2. Add access tokens for web_accessible_resources. The extension will assign 8-9 character tokens and append to the redirectUrl. Tokens expire in 1 second, and requests without a valid token are blocked. 
3. I left support in for Firefox surrogates that need to be skipped if they're going to be blocked by CORS. None of our surrogates need this yet. 

## Steps to test this PR:
1. Fixes issues with redirecting surrogates on sites with strict CSP, test this in both Firefox and Chrome
   - Go to twitter.com, open devtools, check that our surrogate loaded: window.ga.create() should return a Proxy
   - Go to cbs.com, open devtool, try to watch any video, the video should play, window.googletag.pubads() should return a Proxy

2. Test direct access to web resources, Chrome only, this should not work. Accessing web resources requries a valid access token assigned by the extension.
   - Open a new tab, go to `chrome-extension://bkdgflcldnnnapblkhphbgpggdiikppg/web_accessible_resources/analytics.js`
   - It should be blocked since you don't have an access token

## Automated tests:

###### Reviewer Checklist:


###### PR Author Checklist:
